src/config_h.rs.in

```diff
@@ -1,12 +1,25 @@
+
+
 use std::path::Path;
 
-pub fn get_assetsdir() -> &'static str {
+fn get_datadir() -> &'static str {
+  "@prefix@/share/"
+}
+
+pub fn get_assetsdir() -> String {
   if Path::new("./assets/cards").exists() {
-    return ".";
+    return ".".to_owned();
   }
-  "@prefix@/share/@PACKAGE_TARNAME@"
+  format!("{}/{}", get_datadir(), "@PACKAGE_TARNAME@")
 }
 
-pub fn get_localedir() -> &'static str {
-  "@prefix@/share/locale"
+pub fn get_localedir() -> String {
+  format!("{}/{}", get_datadir(), "locale")
+}
+
+pub fn get_pixmapsdir() -> String {
+  if Path::new("./assets").exists() {
+    return "./assets".to_owned();
+  }
+  format!("{}/{}", get_datadir(), "pixmaps")
```

Comment:
only thing I would comment is something like this is more rust-like:
```rust


use std::path::Path;

fn get_datadir() -> &'static str {
  "@prefix@/share/"
}

pub fn get_assetsdir() -> String {
  if Path::new("./assets/cards").exists() {
    ".".to_owned()
  }
  else {
    format!("{}/{}", get_datadir(), "@PACKAGE_TARNAME@")
  }
}

pub fn get_localedir() -> String {
  format!("{}/{}", get_datadir(), "locale")
}

pub fn get_pixmapsdir() -> String {
  if Path::new("./assets").exists() {
    "./assets".to_owned()
  }
  else {
    format!("{}/{}", get_datadir(), "pixmaps")
  }
}
```
But I don't mind to much either way. And I know the lack "return" is considered controversial ;) .

Reply:
I didn't know it's considered controversial by some people. I don't see any point to using code that's not needed and does absolutely nothing.